test/support/case.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Alice.Handlers.Case do
+  def all_replies() do
+    message = receive do
+      {:send_message, %{response: message}} -> message
+    after
+      0 -> nil
+    end
+    case message do
+      nil -> []
+      message -> [message | all_replies()]
+    end
+  end
```

Comment:
When `nil` means something, it's good to make it an atom instead
```suggestion
  def all_replies() do
    message = receive do
      {:send_message, %{response: message}} -> message
    after
      0 -> :no_message_received
    end
    case message do
      :no_message_received -> []
      message -> [message | all_replies()]
    end
  end
```

Reply:
You're right, this is 100% better.